kratos/geometries/brep_curve_on_surface.h

```diff
@@ -338,6 +338,70 @@ class BrepCurveOnSurface
             mCurveNurbsInterval.GetT0(), mCurveNurbsInterval.GetT1());
     }
 
+    ///@}
+    ///@name IsInside
+    ///@{
+
+    /// returns if rPointLocalCoordinates[0] is inside -> 1 or outside -> 0
+    int IsInsideLocalSpace(
+        const CoordinatesArrayType& rPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+    ) const override
+    {
+        const double min_parameter = mCurveNurbsInterval.MinParameter();
+        if (rPointLocalCoordinates[0] < min_parameter) {
+            return 0;
+        }
+        else if (std::abs(rPointLocalCoordinates[0] - min_parameter) < Tolerance) {
+            return 2;
```

Comment:
What 2 means?

Reply:
2 means that a specific point is on the boundary. Which for some methods might be advantageous to know. We decided on this interface quite a long time ago.